### PR TITLE
propagate new bc::settings class

### PR DESCRIPTION
**Pull request by toxeus**

### CODE REVIEW

File: include/bitcoin/blockchain/organizers/header_organizer.hpp [hunk 1]
```diff
@@ -64,7 +65,7 @@ class BCB_API header_organizer
 private:
     // Verify sub-sequence.
     void handle_check(const code& ec, header_const_ptr header,
-        result_handler handler);
+        result_handler handler, const bc::settings& bitcoin_settings);
```
Comment (evoskuil) [inline include/bitcoin/blockchain/organizers/header_organizer.hpp]: Use member here.

File: src/interface/block_chain.cpp [hunk 2]
```diff
@@ -115,7 +118,7 @@ bool block_chain::get_header(chain::header& out_header, size_t height,
 bool block_chain::get_header(chain::header& out_header, size_t& out_height,
     const hash_digest& block_hash, bool block_index) const
 {
-    auto result = database_.blocks().get(block_hash);
+    auto result = database_.blocks().get(block_hash, bitcoin_settings_);
```
Comment (evoskuil) [inline src/interface/block_chain.cpp]: Don’t pass here, use in database construction.

File: include/bitcoin/blockchain/organizers/header_organizer.hpp (hunk 1 above)
Comment (evoskuil) [reply include/bitcoin/blockchain/organizers/header_organizer.hpp]: And all similar below.

Comment (toxeus) [reply include/bitcoin/blockchain/organizers/header_organizer.hpp]: Done.

File: src/interface/block_chain.cpp (hunk 2 above)
Comment (toxeus) [reply src/interface/block_chain.cpp]: Done.

**Comment by evoskuil** [comment]
Rebased: https://github.com/libbitcoin/libbitcoin-blockchain/pull/426

**Pull request closed without merging**